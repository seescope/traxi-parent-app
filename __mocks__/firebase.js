export const profile = {
  UUID: 'YwS0vJ8OE8N6yenxHaV6PdMVLbG3',
  email: 'enakudesu@gmail.com',
  kids: {
    '0': {
      UUID: '2f566920-f598-46d2-8bf2-7bcae115bf0a',
      avatarURL: 'http://i.imgur.com/52mRwuE.jpg',
      deviceType: 'Android',
      name: 'Sam',
      setupID: 5406,
      status: 'INSTALLED',
    },
    '1': {
      UUID: '886b79ea-3572-4b4a-9c25-59b14639ac3d',
      avatarURL: 'https://tiestoclublife.files.wordpress.com/2008/08/dj-tiesto-club-life.jpeg',
      name: 'DJ',
    },
  },
  name: 'Kane Rogers',
  picture: 'https://lh3.googleusercontent.com/-7yt2pBdbMwk/AAAAAAAAAAI/AAAAAAAAD0g/4IHp0iIWDBE/s96-c/photo.jpg',
  token: 'cjj6guKMTME:APA91bFkM3nynjnD5vqxxeYG42t4P6PThP4N1F7VmhNbRW_W_ReCqGtoiDtwG99K4GmD6_Sh_VuCFCR46GwXH314YfdciS0OTGdwe0OGxYvT2RiYhozwcmlLI2TyHn78z26Kw7SYquCD',
};

export const initializeApp = () => {};
export const database = () => ({
  ref: () => ({
    once: () =>
      Promise.resolve({
        val: () => profile,
      }),
  }),
  goOffline: () => {},
});

const firebase = {};

export default firebase;
