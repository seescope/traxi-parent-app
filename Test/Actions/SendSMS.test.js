/* eslint no-native-reassign: "off" */

import sendSMS from '../../App/Actions/SendSMS';
import { mockSet } from 'firebase';

describe('sendSMS', () => {
  it('dispatches a SENDING_SMS action, calls fetch, then dispatches SMS_SENT', () => {
    const TEST_KID = { name: 'Test Name', phoneNumber: '+61401633346' };
    const mockDispatch = jest.fn();

    fetch = jest.fn(() => Promise.resolve());

    return sendSMS(TEST_KID)(mockDispatch).then(() => {
      expect(fetch.mock.calls).toMatchSnapshot();
      expect(mockDispatch.mock.calls).toMatchSnapshot();
      expect(mockSet.mock.calls).toMatchSnapshot();
    });
  });
});
