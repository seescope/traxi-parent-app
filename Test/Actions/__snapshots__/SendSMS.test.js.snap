exports[`sendSMS dispatches a SENDING_SMS action, calls fetch, then dispatches SMS_SENT 1`] = `
Array [
  Array [
    "https://api.twilio.com/2010-04-01/Accounts/ACc04c9d7b6237604d95c65a9d3b991c0c/Messages",
    Object {
      "body": FormData {
        "_parts": Array [
          Array [
            "To",
            "+61401633346",
          ],
          Array [
            "From",
            "traxi",
          ],
          Array [
            "Body",
            "Hi, Test! Welcome to traxi. Tap the following link to get started: http://gettraxi.com/#non-random-uuid",
          ],
        ],
      },
      "headers": Object {
        "Authorization": "Basic QUNjMDRjOWQ3YjYyMzc2MDRkOTVjNjVhOWQzYjk5MWMwYzo5MTNkOTgzMzcwZWVkNTlmMGU0MjhhNmMwZjgwOTYxNg==",
      },
      "method": "POST",
    },
  ],
]
`;

exports[`sendSMS dispatches a SENDING_SMS action, calls fetch, then dispatches SMS_SENT 2`] = `
Array [
  Array [
    Object {
      "type": "SENDING_SMS",
    },
  ],
  Array [
    Object {
      "UUID": "non-random-uuid",
      "type": "SMS_SENT",
    },
  ],
]
`;

exports[`sendSMS dispatches a SENDING_SMS action, calls fetch, then dispatches SMS_SENT 3`] = `
Array [
  Array [
    Object {
      "UUID": "non-random-uuid",
      "deviceType": "unknown",
      "name": "Test Name",
      "phoneNumber": "+61401633346",
      "status": "DOWNLOADING",
    },
    [Function],
  ],
]
`;
